main/src/ca/uwaterloo/flix/language/dbg/printer/LoweredAstPrinter.scala

```diff
@@ -0,0 +1,171 @@
+    case Expression.TypeMatch(exp, rules, tpe, pur, loc) =>
+      val expD = print(exp)
+      val rulesD = rules.map {
+        case LoweredAst.MatchTypeRule(sym, tpe, body) =>
+          val patD = DocAst.Expression.Var(sym)
+          val tpeD = TypePrinter.print(tpe)
+          val bodyD = print(body)
+          (patD, tpeD, bodyD)
+      }
+      DocAst.Expression.TypeMatch(expD, rulesD)
+    case Expression.RelationalChoose(exps, rules, tpe, pur, loc) =>
+      val expD = DocAst.Expression.Tuple(exps.map(print))
+      val rulesD = rules.map {
+        case LoweredAst.RelationalChoiceRule(pat, body) =>
+          val patD = DocAst.Expression.Tuple(pat.map(printRelationalChoicePattern))
+          val guardD = None
+          val bodyD = print(body)
+          (patD, guardD, bodyD)
+      }
+      DocAst.Expression.Match(expD, rulesD)
+    case Expression.Tag(sym, exp, tpe, pur, loc) => DocAst.Expression.Tag(sym.sym, List(print(exp)))
+    case Expression.Tuple(elms, tpe, pur, loc) => DocAst.Expression.Tuple(elms.map(print))
+    case Expression.RecordEmpty(tpe, loc) => DocAst.Expression.RecordEmpty
+    case Expression.RecordSelect(exp, field, tpe, pur, loc) => DocAst.Expression.RecordSelect(field, print(exp))
+    case Expression.RecordExtend(field, value, rest, tpe, pur, loc) => DocAst.Expression.RecordExtend(field, print(value), print(rest))
+    case Expression.RecordRestrict(field, rest, tpe, pur, loc) => DocAst.Expression.RecordRestrict(field, print(rest))
+    case Expression.ArrayLit(exps, exp, tpe, pur, loc) => DocAst.Expression.ArrayLit(exps.map(print)) // TODO needs region
+    case Expression.ArrayNew(exp1, exp2, exp3, tpe, pur, loc) => DocAst.Expression.ArrayNew(print(exp1), print(exp2)) // TODO needs region
+    case Expression.ArrayLoad(base, index, tpe, pur, loc) => DocAst.Expression.ArrayLoad(print(base), print(index))
+    case Expression.ArrayLength(base, pur, loc) => DocAst.Expression.ArrayLength(print(base))
+    case Expression.ArrayStore(base, index, elm, pur, loc) => DocAst.Expression.ArrayStore(print(base), print(index), print(elm))
+    case Expression.VectorLit(exps, tpe, pur, loc) => ???
+    case Expression.VectorLoad(exp1, exp2, tpe, pur, loc) => ???
+    case Expression.VectorLength(exp, loc) => ???
+    case Expression.Ref(exp1, exp2, tpe, pur, loc) => DocAst.Expression.Ref(print(exp1)) // TODO needs region
+    case Expression.Deref(exp, tpe, pur, loc) => DocAst.Expression.Deref(print(exp))
+    case Expression.Assign(exp1, exp2, tpe, pur, loc) => DocAst.Expression.Assign(print(exp1), print(exp2))
+    case Expression.Ascribe(exp, tpe, pur, loc) => DocAst.Expression.Ascription(print(exp), TypePrinter.print(tpe))
+    case Expression.InstanceOf(exp, clazz, loc) => DocAst.Expression.InstanceOf(print(exp), clazz)
+    case Expression.Cast(exp, declaredType, declaredPur, tpe, pur, loc) => DocAst.Expression.Cast(print(exp), TypePrinter.print(declaredType.get)) // TODO needs eff
```

Comment:
I think this is a general todo and shouldn't be fixed here. effects are generally ignored atm